File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/trip/TripDAO.java
package org.craftedsw.tripservicekata.trip;

import java.util.List;

import org.craftedsw.tripservicekata.exception.DependendClassCallDuringUnitTestException;
import org.craftedsw.tripservicekata.user.User;

public class TripDAO {

    /**
     * @deprecated (use instance method)
     */
	public static List<Trip> findTripsByUser(User user) {
		throw new DependendClassCallDuringUnitTestException(
				"TripDAO should not be invoked on an unit test.");
	}

    public List<Trip> findTripsBy(User user) {
        return findTripsByUser(user);
    }
	
}

File: php/src/TripServiceKata/Trip/TripDAO.php
<?php

namespace TripServiceKata\Trip;

use TripServiceKata\User\User;
use TripServiceKata\Exception\DependentClassCalledDuringUnitTestException;

class TripDAO
{
    /**
     * @param User $user
     * @return array
     */
    public static function findTripsByUser(User $user)
    {
        throw new DependentClassCalledDuringUnitTestException('TripDAO should not be invoked on an unit test.');
    }
}
